flutter/lib/src/sentry_flutter.dart

```diff
@@ -67,15 +67,19 @@ mixin SentryFlutter {
     final platformDispatcher = PlatformDispatcher.instance;
     final wrapper = PlatformDispatcherWrapper(platformDispatcher);
 
-    // Flutter Web don't capture [Future] errors if using [PlatformDispatcher.onError] and not
+    // Flutter Web doesn't capture [Future] errors if using [PlatformDispatcher.onError] and not
     // the [runZonedGuarded].
     // likely due to https://github.com/flutter/flutter/issues/100277
-    final isOnErrorSupported = options.platformChecker.isWeb
-        ? false
-        : wrapper.isOnErrorSupported(options);
+    final bool isOnErrorSupported =
+        !options.platformChecker.isWeb && wrapper.isOnErrorSupported(options);
 
-    final runZonedGuardedOnError =
-        options.platformChecker.isWeb ? _createRunZonedGuardedOnError() : null;
+    final bool isRootZone = options.platformChecker.isRootZone;
+
+    // If onError is not supported and no custom zone exists, use runZonedGuarded to capture errors.
+    final bool useRunZonedGuarded = !isOnErrorSupported && isRootZone;
+
+    RunZonedGuardedOnError? runZonedGuardedOnError =
+        useRunZonedGuarded ? _createRunZonedGuardedOnError() : null;
```

Comment:
imo it would be great if we can identify if the user used our `Sentry.runZonedGuarded` vs a custom one and if they use a custom one we log a warning that we recommend switching to our implementation.

Reply:
Hmm, i don't see how we can reliably detect if code is run in the Sentry zone. Do you have an idea?

Reply:
maybe it can be as simple as setting an internal flag, wdyt

Reply:
Added a simple flag. While not perfect, we can't detect that init was called within Sentry.runZonedGuarded, we can at least reasonably assume it.

Reply:
hm, sorry yeah if we can't reliably do it then let's just omit it

Reply:
I guess the docs will be enough for this

Reply:
k, i have reverted this.